pkg/investigations/registry.go

```diff
@@ -0,0 +1,27 @@
+package investigation
+
+// availableInvestigations holds all Investigation implementations.
+var availableInvestigations = []Investigation{}
+
+// RegisterInvestigation registers a new Investigation implementation.
+// This function is called from every init() function in every investigation
+// package. It is done this way to limit the number of places that needs to be
+// edit when adding a new investigation.
+// Future improvement is to use the proper mapping that can return multiple investigations
+// which can make this irrelevant.
+func RegisterInvestigation(inv Investigation) {
+	availableInvestigations = append(availableInvestigations, inv)
+}
+
+// GetInvestigation returns the first Investigation that applies to the given alert title.
+// This is a naive version that only returns the first matching investigation and ignores the rest.
+// Future improvement is to use the proper mapping that can return multiple investigations
+// linked to single alert type.
+func GetInvestigation(title string, experimental bool) Investigation {
+	for _, inv := range availableInvestigations {
+		if inv.CheckActive(title) && (experimental || !inv.IsExperimental()) {
```

Comment:
non-blocking comment
naming things is hard but I will also give it a try :D 
inv.MatchesTitle(title) instead of CheckActive?

Reply:
`HandlesAlert`?